docs/about/architecture.md

```diff
@@ -269,7 +269,85 @@ Coder on Kubernetes.
 - For GCP:
   [Google Cloud Identity Platform](https://cloud.google.com/architecture/identity/single-sign-on)
 
-### Dev Container
+### Air-gapped architecture
+
+The air-gapped deployment model refers to the setup of Coder's development
+environment within a restricted network environment that lacks internet
+connectivity. This deployment model is often required for organizations with
+strict security policies or those operating in isolated environments, such as
+government agencies or certain enterprise setups.
+
+The key features of the air-gapped architecture include:
+
+- _Offline installation_: Deploy workspaces without relying on an external
+  internet connection.
+- _Isolated package/plugin repositories_: Depend on local repositories for
+  software installation, updates, and security patches.
+- _Secure data transfer_: Enable encrypted communication channels and robust
+  access controls to safeguard sensitive information.
+
+Learn more about [offline deployments](../install/offline.md) of Coder.
+
+![Architecture Diagram](../images/architecture-air-gapped.png)
+
+#### Components
+
+The deployment model includes:
+
+- _Workspace provisioners_ with direct access to self-hosted package and plugin
+  repositories and restricted internet access.
+- _Mirror of Terraform Registry_ with multiple versions of Terraform plugins.
+- _Certificate Authority_ with all TLS certificates to build secure
+  communication channels.
+
+The model is compatible with various infrastructure models, enabling deployment
+across multiple regions and diverse cloud platforms.
+
+##### Workload resources
+
+**Workspace provisioner**
+
+- Includes Terraform binary in the container or system image.
+- Checks out Terraform plugins from self-hosted _Registry_ mirror.
+- Deploys workspace images stored in the self-hosted _Container Registry_.
```

Comment:
This is a bit specific to Kubernetes/Docker. It might be good to abstract out the 'container registry' piece to something like an 'internal private cloud'.

Reply:
TBH I wouldn't overcomplicate the diagram with "internal private cloud" and show the relevant components. I bet that "Docker image" approach would be the most popular setup, so maybe keep the sentence here?

Reply:
Fair enough 👍